arch/unitroot/cointegration.py

```diff
@@ -0,0 +1,306 @@
+        n_lags: int, array, None
+            This determines which values the function should search for the
+            best vector.
+
+            * int: If an int, the calculation is done for only that lag
+            * array: If an array of two integers, the first value is where
+                        the search begins and the second is where it ends
+            * None: If None is given, the function searches from 2 to
+                        ceiling of the cube root of the number of observations
+                        divided by two plus two in order to ensure at least
+                        one value is searched.
+                        I.E last_lag = (n_obs**(1/3) / 2) + 2
+
+        trend : str {'c', 'ct'}
+            The trend term included in regression for cointegrating equation.
+
+            * 'c' : constant.
+            * 'ct' : constant and linear trend.
+            * also available quadratic trend 'ctt', and no constant 'nc'.
+
+        normalize: Boolean
+            If true, the first entry in the parameter vector is normalized to
+            one and everything else is divided by the first entry. This is
+            because any cointegrating vector could be multiplied by a scalar
+            and still be a cointegrating vector.
+        reverse: Boolean
+            The series must be ordered from the latest data points to the last.
+            This is in order to calculate the differences. Using this, you can
+            reverse the ordering of your data points.
+
+        Returns
+        -------
+        coefs: array
+            A vector that will create a L(0) time series if a
+            combination sexists.
+
+        Notes
+        -----
+        The data must go from the latest observations to the earliest. If not,
+        the coef vector will be the opposite sign.
+
+        The series should be checked independently for their integration order.
+        The series must be L(1) to get consistent results. You can check this
+        by using the int_order function.
+
+        References
+        ----------
+        .. [1] Stock, J. H., & Watson, M. W. (1993). A simple estimator of
+        cointegrating vectors in higher order integrated systems.
+        Econometrica: Journal of the Econometric Society, 783-820.
+        .. [2] Hamilton, J. D. (1994). Time series analysis
+        (Vol. 2, pp. 690-696). Princeton, NJ: Princeton university press.
+        """
+        self.ic = []
+        self.max_val = []
+        self.model = ''
+        self.coefs = []
+
+        y1 = add_trend(y1, trend=trend, prepend=True)
+        y1 = y1.reset_index(drop=True)
+        if reverse:
+            y0, y1 = y0[::-1], y1[::-1]
+
+        if _is_using_pandas(y0, y1):
+            columns = list(y1.columns)
+
+        else:
+            # Need to check if NumPy, because I can only support those two
+            n_obs, k = y1.shape
+            columns = [f'Var_{x}' for x in range(k)]
```

Comment:
We still have Python 3.5, so need to use old style formatting.

Reply:
Changed to .format()